From 367d56fad11e2156724f183505e030236e9cd0c4 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Tue, 10 Oct 2017 07:24:18 +0200
Subject: [PATCH] temporary patch to attempt fixing setcap

setcap failed because it didn't properly follow DESTDIR, Elvis proposed an
awesome fix for this in D8064. it stands to reason that applying this
should make cap preservation up into the installed system work.
---
 src/start_kdeinit/CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/start_kdeinit/CMakeLists.txt b/src/start_kdeinit/CMakeLists.txt
index 74fddc5..7183de9 100644
--- a/src/start_kdeinit/CMakeLists.txt
+++ b/src/start_kdeinit/CMakeLists.txt
@@ -11,11 +11,17 @@ install(TARGETS start_kdeinit_wrapper DESTINATION ${CMAKE_INSTALL_FULL_LIBEXECDI
 
 if (KDEINIT_OOM_PROTECT)
        if (Libcap_FOUND)
+                         install( CODE "execute_process(
+                          COMMAND
+                                 echo
+                                 CAP_SYS_RESOURCE=+ep
+                                 \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LIBEXECDIR_KF5}/start_kdeinit)"
+                )
                   install( CODE "execute_process(
                           COMMAND
                                  ${SETCAP_EXECUTABLE}
                                  CAP_SYS_RESOURCE=+ep
-                                 $ENV{DESTDIR}${CMAKE_INSTALL_FULL_LIBEXECDIR_KF5}/start_kdeinit)"
+                                 \$ENV{DESTDIR}${CMAKE_INSTALL_FULL_LIBEXECDIR_KF5}/start_kdeinit)"
                 )
        else()
                   install(CODE "
-- 
2.7.4

